ocrs/src/recognition.rs

```diff
@@ -537,10 +552,26 @@ impl TextRecognizer {
         // batching and parallel processing. Re-sort them into input order.
         line_rec_results.sort_by_key(|result| result.line.index);
 
-        let text_lines = text_lines_from_recognition_results(&line_rec_results, &alphabet);
+        let text_lines = text_lines_from_recognition_results(&line_rec_results, alphabet);
 
         Ok(text_lines)
     }
+
+    fn filter_excluded_char_labels<'a>(
+        excluded_char_labels: Option<&[usize]>,
+        input_seq_slice: &'a mut NdTensorViewMut<'_, f32, 2>,
```

Comment:
`NdTensorViewMut` is cheap to pass by value (it is a slice plus 2 `usize` values per dimension), so you can change this to `NdTensorViewMut<'a, f32, 2>`. Either way is fine though.

Reply:
Yeah, I tried passing `NdTensorViewMut` earlier but ran into lifetime issue.
My earlier code was  
```
 fn filter_excluded_char_labels<'a>(
        excluded_char_labels: Option<&[usize]>,
        mut input_seq_slice: NdTensorViewMut<'a, f32, 2>,
    ) -> NdTensorView<'a, f32, 2> {
        if let Some(excluded_char_labels) = excluded_char_labels {
            for row in 0..input_seq_slice.shape()[0] {
                for &excluded_char_label in excluded_char_labels.iter() {
                    // Setting the output value of excluded char to -Inf causes the
                    // `decode_method` to favour chars other than the excluded char.
                    input_seq_slice[[row, excluded_char_label]] = f32::NEG_INFINITY;
                }
            }
        }
        input_seq_slice.view()
    }
```  
Which cause the following Compiler error: 
```
error[E0515]: cannot return value referencing function parameter `input_seq_slice`
   --> ocrs/src/recognition.rs:573:9
    |
573 |         input_seq_slice.view()
    |         ---------------^^^^^^^
    |         |
    |         returns a value referencing data owned by the current function
    |         `input_seq_slice` is borrowed here

```  
 The pushed code was work around for this issue. Can you suggest what should be the function signature in this case ?

Reply:
Ah, I see. The problem is that `.view()` creates an immutable view which borrows from, and thus cannot outlive the caller. It's signature is `fn view(&self) -> NdTensorView<'_, ...>`. You'd need a `fn into_view(self) -> NdTensorView<'a, ...>` method instead.

It's fine to stick with the code as it currently is. An alternative pattern would be to do something like this:

```rs
let mut input_seq_slice = ...;

filter_excluded_char_labels(
  excluded_labels,
  input_seq_slice.view_mut() /* create a new, temporary mutable view */
);

// Use original `input_seq_slice` as before.
```

Reply:
sticking with the original in this one.